File: LLM-Translations/GPT-4/codenet/C++/Go/s526060174.go

package main

import (
	"fmt"
)

func main() {
	var n, m int
	var k, a int
	var x [31]int
	fmt.Scan(&n, &m)
	for i := 0; i < n; i++ {
		fmt.Scan(&k)
		for j := 0; j < k; j++ {
			fmt.Scan(&a)
			x[a]++
		}
	}
	ans := 0
	for i := 1; i <= m; i++ {
		if x[i] == n {
			ans++
		}
	}
	fmt.Println(ans)
}

// 

File: LLM-Translations/GPT-4/codenet/C++/Python/s526060174.py

from sys import stdin

def main():
    n, m = map(int, input().split())
    x = [0] * 31
    for _ in range(n):
        k, *a = map(int, input().split())
        for j in a:
            x[j] += 1
    ans = 0
    for i in range(1, m+1):
        if x[i] == n:
            ans += 1
    print(ans)

if __name__ == "__main__":
    main()

#

